streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -331,9 +307,9 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
                     log.warn("Restoring GlobalStore {} failed due to: {}. Deleting global store to recreate from scratch.",
                         storeName,
                         recoverableException.toString());
-                    reinitializeStateStoresForPartitions(recoverableException.partitions(), globalProcessorContext);
 
-                    stateRestoreListener.onRestoreStart(topicPartition, storeName, offset, highWatermark);
+                    // TODO K9113: we remove the re-init logic and push it to be handled by the thread directly
```

Comment:
There's still a TODO here that hasn't been handled.

Reply:
Yes it is to be handled in a follow-up PR.